### PR TITLE
feat: templ.Join method renders multiple components into a single component

### PR DESCRIPTION
**Pull request by Quarkz14**
Heyoo, 
I have a PR for the proposal suggested in this [issue](https://github.com/a-h/templ/issues/788) I followed the example solution you given @a-h. I have updated the docs too to explain the new Join method with an example in the `Temple Composition` section. Let me know what you think. I will rename the commit message as part of any changes in the feedback

### CODE REVIEW

File: join.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package templ
+
+import (
+	"context"
+	"io"
+)
+
+// Pass any number of templ.Components to get a single templ.Component with the components rendered
```
Comment (joerdav) [inline join.go:8]: Go function comments should begin with the name of the documented function.

```suggestion
// Join returns a single `templ.Component` that will render provided components in order.
// If any of the components return an error the Join component will immediately return with the error.
```

File: docs/docs/03-syntax-and-usage/10-template-composition.md [hunk 2]
```diff
@@ -235,6 +235,49 @@ func main() {
 	<p>Dynamic contents</p>
 </div>
 ```
+## Joining Components
+With your templ components you can use `templ.Join` to generate a single `templ.Component` from your components.
+```templ
```
Comment (joerdav) [inline docs/docs/03-syntax-and-usage/10-template-composition.md:240]: Tabs not spaces ;)

Also I think that joining generated templ components here would be more intuitive for the docs:

templates.go
```templ
package main

templ hello() {
	<span>hello</span>
}

templ world() {
	<span>world</span>
}

templ helloWorld() {
	@templ.Join(hello(), world())
}
```

main.go
```go
package main

import (
	"context"
	"os"
)

func main() {
	helloWorld().Render(context.Background(), os.Stdout)
}
```

Output:
```html
<span>hello</span><span>world</span>
```

File: docs/docs/03-syntax-and-usage/10-template-composition.md [hunk 3]
```diff
@@ -235,6 +235,49 @@ func main() {
 	<p>Dynamic contents</p>
 </div>
 ```
+## Joining Components
+With your templ components you can use `templ.Join` to generate a single `templ.Component` from your components.
```
Comment (joerdav) [inline docs/docs/03-syntax-and-usage/10-template-composition.md:239]:
```suggestion
Components can be aggregated into a single Component using `templ.Join`.
```

File: join_test.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		}
+		return nil
+	})
+	world := templ.ComponentFunc(func(ctx context.Context, w io.Writer) error {
+		if _, err := io.WriteString(w, "World"); err != nil {
+			t.Fatalf("failed to write string: %v", err)
+		}
+		return nil
+	})
+	components := []templ.Component{hello, world}
```
Comment (joerdav) [inline join_test.go:26]: These definitions would probably be cleaner defined in the table test directly.

**Review by joerdav — COMMENTED** [review]
Thanks for the contribution and time!! A few small nits here and there I spotted.

File: join_test.go (hunk 4 above)
Comment (Quarkz14) [reply join_test.go]: I initialise them in the test structs now.

**Review by a-h — APPROVED** [review]

**Pull request merged**